YubiKit/YubiKit/Sessions/Shared/Services/OATH/YKFKeyOATHService.m

```diff
@@ -112,6 +112,21 @@ - (void)executeDeleteRequest:(YKFKeyOATHDeleteRequest *)request completion:(YKFK
     }];
 }
 
+- (void)executeRenameRequest:(YKFKeyOATHRenameRequest *)request completion:(YKFKeyOATHServiceCompletionBlock)completion {
+    YKFParameterAssertReturn(request);
+    YKFParameterAssertReturn(completion);
+    
+    YKFKeySessionError *credentialError = [YKFOATHCredentialValidator validateCredential:request.credential includeSecret:NO];
+    if (credentialError) {
+        completion(credentialError);
+    }
+    
+    [self executeOATHRequest:request completion:^(NSData * _Nullable result, NSError * _Nullable error) {
```

Comment:
IMHO you need to check firmware version before calling rename on the key, because this operation is supported only on 5.3+ and provide proper error message to developer.

Reply:
You will get a `YKFKeyAPDUErrorCodeInsNotSupported` if you try to rename credentials on a older key.

Reply:
It's preferable that we handle all expected APDU error codes within SDK and provide only YKFKeyOATHError error codes + descriptions to developers. At least that's how current API do...